src/librustc_codegen_ironox/type_of.rs

```diff
@@ -10,17 +17,141 @@ pub trait LayoutIronOxExt<'tcx> {
                              scalar: &layout::Scalar) -> Type;
 }
 
+/// Return the IronOx `Type`s of the fields of the specified `layout`.
+fn struct_field_types(
+    cx: &CodegenCx<'_, 'tcx>,
+    layout: TyLayout<'tcx>) -> Vec<Type> {
+    let mut fields = vec![];
+    for i in layout.fields.index_by_increasing_offset() {
+        let field = layout.field(cx, i);
+        // FIXME: handle field alignment
+        fields.push(field.ironox_type(cx));
+    }
+    fields
+}
+
 impl LayoutIronOxExt<'tcx> for TyLayout<'tcx> {
     fn ironox_type(&self, cx: &CodegenCx<'_, 'tcx>) -> Type {
-        unimplemented!("ironox_type");
+        if let layout::Abi::Scalar(ref scalar) = self.abi {
```

Comment:
Can you explain the purpose of the `ironox_type` method here?

Reply:
It returns the `Type` of a Rust type. Used here:

https://github.com/gabi-250/rust/blob/508c80943387801b1f3d2701b8562652d3f6ab1e/src/librustc_codegen_ironox/type_.rs#L255-L257

and here:

https://github.com/gabi-250/rust/blob/508c80943387801b1f3d2701b8562652d3f6ab1e/src/librustc_codegen_ironox/type_.rs#L259-L261

`Type` is a type I defined for my backend. All codegen backends have to define their own `Type` (they don't operate directly on Rust types such as `TyLayout`).

Basically, it transofrms a `TyLayout` (Rust type) into a `Type` (backend-specific type).